src/data/int/basic.lean

```diff
@@ -1069,6 +1072,20 @@ units.ext_iff.1 $ nat.units_eq_one ⟨nat_abs u, nat_abs ↑u⁻¹,
 theorem units_eq_one_or (u : units ℤ) : u = 1 ∨ u = -1 :=
 by simpa only [units.ext_iff, units_nat_abs] using nat_abs_eq u
 
+lemma _root_.is_unit.eq_one_or {a : ℤ} : is_unit a → a = 1 ∨ a = -1
+| ⟨x, hx⟩ := hx ▸ (units_eq_one_or _).imp (congr_arg coe) (congr_arg coe)
```

Comment:
Thinking again, either we should rename `int.units_eq_one_or` to `units.eq_one_or`, or call the new lemma `int.is_unit_eq_one_or`. Thoughts, @jcommelin, since you suggested the dot notation?

Reply:
I think maybe the suggestion of dot notation wasn't that great. It feels a little bit forced. So `int.is_unit_eq_one_or` seems good to me.

Reply:
I made the change!